### PR TITLE
Optimize function name and comment of merge join for sorted inputs

### PR DESCRIPTION
**Pull request by kewang1024**
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/FastPathMergeJoinOptimizer.java [hunk 1]
```diff
@@ -35,13 +35,13 @@
 import static com.google.common.collect.ImmutableList.toImmutableList;
 import static java.util.Objects.requireNonNull;
 
-public class MergeJoinOptimizer
+public class FastPathMergeJoinOptimizer
```
Comment (rschlussel) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/FastPathMergeJoinOptimizer.java:38]: similarly would rename this that it's for merge join on pre-sorted inputs

File: presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/FastPathMergeJoinOptimizer.java [hunk 2]
```diff
@@ -114,7 +116,7 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
             return node;
         }
 
-        private boolean isMergeJoinEligible(PlanNode left, PlanNode right, JoinNode node)
+        private boolean meetDataRequirement(PlanNode left, PlanNode right, JoinNode node)
```
Comment (rschlussel) [inline presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/FastPathMergeJoinOptimizer.java:119]: meet -> meets

File: presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java [hunk 3]
```diff
@@ -1273,7 +1273,7 @@ public SystemSessionProperties(
                         featuresConfig.isStreamingForPartialAggregationEnabled(),
                         false),
                 booleanProperty(
-                        PREFER_MERGE_JOIN,
+                        PREFER_MERGE_JOIN_FOR_SORTED_INPUTS,
                         "Prefer merge join for sorted join inputs, e.g., tables pre-sorted, pre-partitioned by join columns." +
                                 "To make it work, the connector needs to guarantee and expose the data properties of the underlying table.",
                         featuresConfig.isPreferMergeJoin(),
```
Comment (rschlussel) [inline presto-main/src/main/java/com/facebook/presto/SystemSessionProperties.java:1279]: you need to change the config property name too

**Review by rschlussel — APPROVED** [review]

**Review by amitkdutta — APPROVED** [review]
Looks good. Thanks @kewang1024

**Pull request merged**